File: backend/src/main/java/com/school/management/entity/Student.java
package com.school.management.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDate;

/**
 * 学生实体
 *
 * @author system
 * @since 1.0.0
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("students")
public class Student extends BaseEntity {

    /**
     * 学号
     */
    private String studentNo;

    /**
     * 证件类型
     */
    private String idCardType;

    /**
     * 民族
     */
    private String ethnicity;

    /**
     * 政治面貌
     */
    private String politicalStatus;

    /**
     * 真实姓名(来自users表,非students表字段)
     */
    @TableField(exist = false)
    private String realName;

    /**
     * 性别: 1男 2女 (来自users表,非students表字段)
     */
    @TableField(exist = false)
    private Integer gender;

    /**
     * 手机号(来自users表,非students表字段)
     */
    @TableField(exist = false)
    private String phone;

    /**
     * 身份证号(来自users表,非students表字段)
     */
    @TableField(exist = false)
    private String idCard;

    /**
     * 毕业学校
     */
    @TableField(exist = false)
    private String graduatedSchool;

    /**
     * 家庭住址
     */
    private String homeAddress;

    /**
     * 户口所在地-省
     */
    private String hukouProvince;

    /**
     * 户口所在地-市
     */
    private String hukouCity;

    /**
     * 户口所在地-区
     */
    private String hukouDistrict;

    /**
     * 户口详细地址
     */
    private String hukouAddress;

    /**
     * 户口性质(农业/非农业)
     */
    private String hukouType;

    /**
     * 邮政编码
     */
    private String postalCode;

    /**
     * 是否建档立卡(0否1是)
     */
    private Integer isPovertyRegistered;

    /**
     * 资助申请类型
     */
    private String financialAidType;

    /**
     * 班级ID
     */
    private Long classId;

    /**
     * 年级ID(关联grades表)
     */
    private Long gradeId;

    /**
     * 专业ID
     */
    private Long majorId;

    /**
     * 专业方向ID
     */
    private Long majorDirectionId;

    /**
     * 专业级别/层次(如:中专/大专)
     */
    private String educationLevel;

    /**
     * 学制(如:3年)
     */
    private String studyLength;

    /**
     * 学历(如:中专/大专)
     */
    private String degreeType;

    /**
     * 入学日期
     */
    private LocalDate admissionDate;

    /**
     * 毕业日期
     */
    private LocalDate graduationDate;

    /**
     * 入学层次(如:初中/高中,非数据库字段)
     */
    @TableField(exist = false)
    private String entryLevel;

    /**
     * 学制(如:3年/4年,非数据库字段)
     */
    @TableField(exist = false)
    private String educationSystem;

    /**
     * 学生状态: 0在读 1休学 2退学 3毕业 4转学
     */
    private Integer studentStatus;

    /**
     * 宿舍ID
     */
    private Long dormitoryId;

    /**
     * 床位号
     */
    @TableField("bed_number")
    private String bedNumber;

    /**
     * 监护人姓名
     */
    private String guardianName;

    /**
     * 监护人电话
     */
    private String guardianPhone;

    /**
     * 监护人关系
     */
    private String guardianRelation;

    /**
     * 父亲姓名
     */
    private String fatherName;

    /**
     * 父亲身份证号
     */
    private String fatherIdCard;

    /**
     * 父亲电话
     */
    private String fatherPhone;

    /**
     * 母亲姓名
     */
    private String motherName;

    /**
     * 母亲身份证号
     */
    private String motherIdCard;

    /**
     * 母亲电话
     */
    private String motherPhone;

    /**
     * 监护人身份证号
     */
    private String guardianIdCard;

    /**
     * 用户ID(关联users表)
     */
    private Long userId;

    // ========== 关联字段（非数据库字段） ==========

    /**
     * 班级信息(关联查询)
     */
    @TableField(exist = false)
    private Class classInfo;

    /**
     * 班级名称(关联查询)
     */
    @TableField(exist = false)
    private String className;

    /**
     * 专业名称(关联查询)
     */
    @TableField(exist = false)
    private String majorName;

    /**
     * 年级名称(关联查询)
     */
    @TableField(exist = false)
    private String gradeName;

    /**
     * 宿舍信息(关联查询)
     */
    @TableField(exist = false)
    private Dormitory dormitory;

    /**
     * 宿舍楼号(关联查询)
     */
    @TableField(exist = false)
    private String buildingNo;

    /**
     * 宿舍房间号(关联查询)
     */
    @TableField(exist = false)
    private String roomNo;
}

File: frontend/src/types/student.ts
/**
 * 学生管理类型定义 - DDD架构适配
 */

// ==================== 枚举类型 ====================

/**
 * 学生状态
 */
export type StudentStatus = 0 | 1 | 2 | 3 | 4

export const StudentStatusMap: Record<StudentStatus, string> = {
  0: '在读',
  1: '休学',
  2: '退学',
  3: '毕业',
  4: '转学'
}

/**
 * 性别
 */
export type Gender = 1 | 2

export const GenderMap: Record<Gender, string> = {
  1: '男',
  2: '女'
}

// ==================== 实体类型 ====================

/**
 * 学生实体 - V2 DDD架构字段名
 */
export interface Student {
  id: number | string
  studentNo: string
  userId?: number | string
  username?: string

  // V2 使用 name 代替 realName
  name: string
  phone?: string
  email?: string
  gender: Gender
  // V2 使用 genderText 代替 genderName
  genderText?: string
  idCard?: string
  idCardType?: string
  ethnicity?: string
  politicalStatus?: string
  birthDate?: string
  nativePlace?: string
  avatarUrl?: string

  // 班级信息 (gradeId/majorId/majorDirectionId are on class, not student)
  orgUnitId?: number | string
  className?: string
  gradeName?: string
  gradeLevel?: number
  majorName?: string

  // 学籍信息
  educationLevel?: string
  studyLength?: string
  degreeType?: string
  // V2 使用 enrollmentDate 代替 admissionDate
  enrollmentDate?: string
  expectedGraduationDate?: string
  // V2 使用 status 代替 studentStatus
  status: StudentStatus
  // V2 使用 statusText 代替 studentStatusName
  statusText?: string

  // 宿舍信息已统一到 place_occupants 管理体系
  // 通过 /v9/places API 查询学生入住信息

  // 家庭信息
  homeAddress?: string
  hukouProvince?: string
  hukouCity?: string
  hukouDistrict?: string
  hukouAddress?: string
  hukouType?: string
  postalCode?: string

  // 紧急联系人 (V2 使用)
  emergencyContact?: string
  emergencyPhone?: string

  // 监护人信息 (兼容扩展字段)
  guardianName?: string
  guardianPhone?: string
  guardianRelation?: string
  guardianIdCard?: string
  fatherName?: string
  fatherIdCard?: string
  fatherPhone?: string
  motherName?: string
  motherIdCard?: string
  motherPhone?: string

  // 其他信息
  isPovertyRegistered?: number
  financialAidType?: string
  healthStatus?: string
  allergies?: string
  specialNotes?: string
  remark?: string

  // 时间戳
  createdAt?: string
  updatedAt?: string
}

// ==================== 请求类型 ====================

/**
 * 创建学生请求 - V2 DDD架构字段名
 */
export interface CreateStudentRequest {
  studentNo?: string
  username?: string
  password?: string
  // V2 使用 name 代替 realName
  name: string
  phone?: string
  email?: string
  gender: Gender
  idCard?: string
  idCardType?: string
  ethnicity?: string
  politicalStatus?: string
  birthDate?: string
  nativePlace?: string

  // 班级信息 (gradeId/majorId/majorDirectionId are on class, not student)
  orgUnitId?: number | string

  // 学籍信息
  educationLevel?: string
  studyLength?: string
  degreeType?: string
  // V2 使用 enrollmentDate 代替 admissionDate
  enrollmentDate?: string
  expectedGraduationDate?: string
  // V2 使用 status 代替 studentStatus
  status?: StudentStatus
  entryLevel?: string
  educationSystem?: string
  graduatedSchool?: string

  // 家庭信息
  homeAddress?: string
  hukouProvince?: string
  hukouCity?: string
  hukouDistrict?: string
  hukouAddress?: string
  hukouType?: string
  postalCode?: string

  // 紧急联系人
  emergencyContact?: string
  emergencyPhone?: string

  // 监护人信息
  guardianName?: string
  guardianPhone?: string
  guardianRelation?: string
  guardianIdCard?: string
  fatherName?: string
  fatherIdCard?: string
  fatherPhone?: string
  motherName?: string
  motherIdCard?: string
  motherPhone?: string

  // 其他信息
  isPovertyRegistered?: number
  financialAidType?: string
  healthStatus?: string
  allergies?: string
  specialNotes?: string
  remark?: string
}

/**
 * 更新学生请求
 */
export interface UpdateStudentRequest extends Partial<CreateStudentRequest> {
  id?: number | string
}

/**
 * 学生查询参数 - V2 DDD架构字段名
 */
export interface StudentQueryParams {
  studentNo?: string
  // V2 使用 name 代替 realName (keyword 用于综合搜索)
  name?: string
  keyword?: string
  orgUnitId?: number | string
  gradeId?: number | string // UI-only: used to filter class dropdown, not a student field
  gradeLevel?: number
  // V2 使用 status 代替 studentStatus
  status?: StudentStatus
  gender?: Gender
  enrollmentDateStart?: string
  enrollmentDateEnd?: string
  pageNum?: number
  pageSize?: number
  orgUnitIds?: (number | string)[]
  orgUnitIds?: (number | string)[]
  selfUserId?: number | string
}

/**
 * 学生搜索参数
 */
export interface StudentSearchParams {
  keyword: string
  orgUnitId?: number | string
  limit?: number
}

/**
 * 重置密码请求
 */
export interface ResetPasswordRequest {
  newPassword: string
}

// ==================== 学籍异动记录 ====================

/**
 * 学籍异动记录
 */
export interface StudentStatusChange {
  id: number
  studentId: number
  studentNo?: string
  studentName?: string
  changeType: string
  fromStatus?: string
  toStatus?: string
  fromClassId?: number
  fromClassName?: string
  toClassId?: number
  toClassName?: string
  reason?: string
  effectiveDate?: string
  operatorId?: number
  operatorName?: string
  remark?: string
  createdAt: string
}

/**
 * 异动类型映射
 */
export const ChangeTypeMap: Record<string, string> = {
  ENROLL: '入学',
  SUSPEND: '休学',
  RESUME: '复学',
  GRADUATE: '毕业',
  WITHDRAW: '退学',
  EXPEL: '开除',
  TRANSFER_CLASS: '转班',
  TRANSFER_MAJOR: '转专业'
}

// ==================== 统计类型 ====================

/**
 * 学生统计
 */
export interface StudentStatistics {
  total: number
  active: number
  suspended: number
  withdrawn: number
  graduated: number
  transferred: number
  byGender: {
    male: number
    female: number
  }
  byGrade: Record<string, number>
  byClass: Record<string, number>
}
